test/e2e/fixture/fixture.go

```diff
@@ -1034,6 +1034,12 @@ func EnsureCleanState(t *testing.T, opts ...TestOption) {
 			if err != nil {
 				return err
 			}
+			// Configure git to create files with more permissive permissions to avoid
+			// issues when cleaning up. By default git creates object files as 0444.
+			_, err = Run(repoDirectory(), "git", "config", "core.sharedRepository", "0666")
+			if err != nil {
+				return err
+			}
```

Comment:
This is necessary because for the first time we have a test that writes changes _back_ to the git repo. If we don't add this, then the test fixture (running under a different user) can't clean up the temp dir.